Library/DiscUtils.Vhd/DiskImageFile.cs

```diff
@@ -658,7 +658,7 @@ private void ReadFooter(bool fallbackToFront)
                 }
 
                 _fileStream.Position = 0;
-                StreamUtilities.ReadFully(_fileStream, sector, 0, Sizes.Sector);
+                StreamUtilities.ReadExact(_fileStream, sector, 0, Sizes.Sector);
```

Comment:
Did not throw, does now. Bytes read is not tracked, and Footer.FromBytes() expects a full buffer, so the correct behavior is to throw on a short read.